src/checkbox/checkbox.tsx

```diff
@@ -11,28 +18,23 @@ export const Checkbox = ({
     displaySize = "default",
     ...otherProps
 }: CheckboxProps): JSX.Element => {
-    const [selected, setSelected] = useState<boolean>(checked);
-
-    // =============================================================================
-    // EFFECTS
-    // =============================================================================
-    useEffect(() => {
-        setSelected(checked);
-    }, [checked]);
-
     // =============================================================================
     // EVENT HANDLERS
     // =============================================================================
     const handleOnCheck = (
```

Comment:
as much as possible, we want to reuse the events from native elements. keep the `onChange` handler on the input element. with the css changes above, the icons will not need to manage a click event